src/game/AI/ScriptDevAI/scripts/outland/shattrath_city.cpp

```diff
@@ -699,6 +704,257 @@ bool QuestRewarded_npc_adal(Player* player, Creature* creature, Quest const* que
+            }
+            Unit* crVeteran = m_creature->GetMap()->GetUnit(veteran);
+            if (crVeteran && crVeteran->IsAlive() && crVeteran->AI())
+            {
+                crVeteran->AI()->SetRootSelf(false);
+                crVeteran->GetMotionMaster()->MovePoint(3, veteranExitPosition, FORCED_MOVEMENT_RUN);
+            }
+        }
+        else
+        {
+            for (int x=0; x<4; x++)
+            {
+                for (int y=0; y<4; y++)
+                {
+                    static const std::vector<uint32> entries = {NPC_F_BLOODELF_TRAINEE, NPC_F_DRAENEI_TRAINEE, NPC_M_BLOODELF_TRAINEE, NPC_M_DRAENEI_TRAINEE};
+                    TempSpawnSettings spawnSettings;
+                    spawnSettings.x = recruitEntryPositions[x][y].x;
+                    spawnSettings.y = recruitEntryPositions[x][y].y;
+                    spawnSettings.z = recruitEntryPositions[x][y].z;
+                    spawnSettings.ori = recruitEntryPositions[x][y].o;
+                    spawnSettings.spawnType = TEMPSPAWN_CORPSE_DESPAWN;
+                    spawnSettings.entry = entries[urand(0,3)];
+                    spawnSettings.activeObject = true;
+                    spawnSettings.spawner = m_creature;
+                    Creature* summoned = m_creature->SummonCreature(spawnSettings, m_creature->GetMap());
+                    recruitMatrix[x][y] = summoned->GetObjectGuid();
+                    summoned->GetMotionMaster()->MovePoint(2, recruitEventPositions[x][y], FORCED_MOVEMENT_RUN);
+                }
+            }
+            TempSpawnSettings spawnSettings;
+            spawnSettings.x = veteranEntryPosition.x;
+            spawnSettings.y = veteranEntryPosition.y;
+            spawnSettings.z = veteranEntryPosition.z;
+            spawnSettings.ori = veteranEntryPosition.o;
+            spawnSettings.spawnType = TEMPSPAWN_CORPSE_DESPAWN;
+            spawnSettings.entry = NPC_BLOODELF_VETERAN;
+            spawnSettings.activeObject = true;
+            spawnSettings.spawner = m_creature;
+            Creature* summoned = m_creature->SummonCreature(spawnSettings, m_creature->GetMap());
+            veteran = summoned->GetObjectGuid();
+            summoned->GetMotionMaster()->MovePoint(2, veteranEventPosition, FORCED_MOVEMENT_RUN);
+        }
+    }
+
+    void SummonedMovementInform(Creature* summoned, uint32 motionType, uint32 uiPointId) override
+    {
+        static const std::unordered_set<uint32> entries = {NPC_F_BLOODELF_TRAINEE, NPC_F_DRAENEI_TRAINEE, NPC_M_BLOODELF_TRAINEE, NPC_M_DRAENEI_TRAINEE, NPC_BLOODELF_VETERAN};
+        static const float forward = 5.497790;
+        if (entries.find(summoned->GetEntry()) != entries.end())
+        {
+            switch (uiPointId) {
+                case 2: summoned->GetMotionMaster()->Clear(); summoned->GetMotionMaster()->MoveIdle(); summoned->SetFacingTo(forward); break;
+                case 3: summoned->ForcedDespawn();
+            }
+        }
+    }
+};
```

Comment:
all of this can probably also be solved with SpawnGroups and hard-coded paths
I need to do more research/experimenting before I can confidently move this to SpawnGroups

Reply:
May require access to worldstates tho which are currently not exposed to ACID or dbscripts.